### DISCUSSION TITLE
useSubscription watch - URQL React

### DISCUSSION THREAD

**Discussion by ragokan** [urql-graphql/urql#2338] [Q&A] [answered]
Hey, it might be a weird question, but is there any way to watch the subscription while using **useSubscription**?

I can do something like this but I need something different, just like the vanilla one that uses Wonka. 
```ts
useEffect(() => {
   console.log(newSubscription);
 }, [newSubscription])
```

Basically I want to use it like Wonka while using *useSubscription*.

**Comment by kitten** [comment, marked as answer]
I'm not quite sure what you're trying to do but you can always just start a subscription, manually using the core's client: https://formidable.com/open-source/urql/docs/basics/core/

```js
import { useClient } from 'urql'
import { subscribe, pipe } from 'wonka'

const Comp = () => {
  const client = useClient()
  
  useEffect(() => {
    return pipe(
      client.subscription(query, variables),
      subscribe((result) => { /*...*/ }),
    ).unsubscribe
  }, [client])
}
```

**Reply by ragokan** [reply]
Well, I can actually do this but I just want to ask if I can pipe it with the *useSubscription*?

```ts
  const [response] = useSubscription(WatchCounterDocument); // watch this response, on change call a function
```
